pkg/virtctl/console/console.go

```diff
@@ -69,41 +76,48 @@ func usage() string {
 	return usage
 }
 
-func (c *Console) Run(args []string) error {
-	namespace, _, err := c.clientConfig.Namespace()
-	if err != nil {
-		return err
-	}
+func (c *Console) run(cmd *cobra.Command, args []string) error {
 
 	vmi := args[0]
 
-	virtCli, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
-	if err != nil {
+	var err error
+
+	if c.namespace, _, err = c.clientConfig.Namespace(); err != nil {
 		return err
 	}
 
-	stdinReader, stdinWriter := io.Pipe()
-	stdoutReader, stdoutWriter := io.Pipe()
+	if c.client, err = kubecli.GetKubevirtClientFromClientConfig(c.clientConfig); err != nil {
+		return fmt.Errorf("cannot obtain KubeVirt client: %v", err)
+	}
 
 	// in -> stdinWriter | stdinReader -> console
 	// out <- stdoutReader | stdoutWriter <- console
 	// Wait until the virtual machine is in running phase, user interrupt or timeout
+	c.stdinReader, c.stdinWriter = io.Pipe()
+	c.stdoutReader, c.stdoutWriter = io.Pipe()
```

Comment:
Move definition of these to `handleConsoleConnection` instead perhaps? Allows to remove them from the command's struct as well.